File: cpp/1732. 找到最高海拔.cpp
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int n=gain.size();
        int maxNum=0,sum=0;
        for(int i=0;i<n;i++){
            sum+=gain[i];
            if(sum>maxNum){
                maxNum=sum;
            }
        }
        return maxNum;
    }
};

File: 刷题/golang/1732. 找到最高海拔.go
func largestAltitude(gain []int) (ans int) {
    total := 0
    for _, x := range gain {
        total += x
        ans = max(ans, total)
    }
    return
}

func max(a, b int) int {
    if b > a {
        return b
    }
    return a
}